web/client-api/src/main/java/io/deephaven/web/client/api/i18n/JsDateTimeFormat.java

```diff
@@ -237,7 +237,7 @@ private String formatAsLongNanos(long unwrapped, JsTimeZone timeZone) {
             // append required digits (may truncate the number if requested)
             String nanoString = Integer.toString(nanos);
             assert nanoString.length() <= 9;
-            for (int i = nanoString.length(); i < 9; i++) {
+            for (int i = 0; i < 9 - nanoString.length() && i < nanoCount; i++) {
```

Comment:
Do we care about calling `nanoString.length()` for each iteration of the loop? I don't know if there are extra JS associated costs calling string length function.

Reply:
We do not, like Java, JS strings are immutable, so this value is cached in some way.

Also, max of 8 iterations through this loop - the emulated long math in the mod and divide operations above will likely dwarf this.